### PR TITLE
fix: allow_hyphen_values for constructor args

### PR DESCRIPTION
**Pull request by mattsse**
clap doesn't accep `-<int>` and requires escaped `'-<int>'` so we have to manually unescape here

### CODE REVIEW

File: crates/forge/bin/cmd/create.rs [hunk 1]
```diff
@@ -380,9 +380,17 @@ impl CreateArgs {
             let ty = input
                 .resolve()
                 .wrap_err_with(|| format!("Could not resolve constructor arg: input={input}"))?;
+
+            // negative ints must be escaped on CLI, so we strip '
+            let arg = if matches!(ty, DynSolType::Int(_)) {
+                arg.trim_start_matches("'").trim_end_matches("'")
+            } else {
+                arg.as_str()
+            };
```
Comment (DaniPopes) [inline crates/forge/bin/cmd/create.rs:389]: don't really like this, could passing through `-- -1` work?

Comment (mattsse) [reply crates/forge/bin/cmd/create.rs:389]: no calp just interprets `-<val>` as an arg not value

> error: unexpected argument '-5' found

Comment (DaniPopes) [reply crates/forge/bin/cmd/create.rs:389]: This should do it then on `constructor_args`: https://docs.rs/clap/latest/clap/struct.Arg.html#method.allow_hyphen_values

Comment (mattsse) [reply crates/forge/bin/cmd/create.rs:389]: cool TIL

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**